### PR TITLE
Fix: socks5 usernames and passwords can BOTH be up to 255 bytes

### PR DESCRIPTION
**Pull request by Yawning**
This should be rather self-explanatory.  The existing authentication routine limits the entire username/password request to 255 bytes, however that limit applies to the UNAME/PASSWD fields individually.  My PR also enforces the RFC specified minimum of 1 byte.

https://github.com/xjasonlyu/tun2socks/blob/main/transport/socks5/rfc1929.txt#L40
```
2.  Initial negotiation

   Once the SOCKS V5 server has started, and the client has selected the
   Username/Password Authentication protocol, the Username/Password
   subnegotiation begins.  This begins with the client producing a
   Username/Password request:

           +----+------+----------+------+----------+
           |VER | ULEN |  UNAME   | PLEN |  PASSWD  |
           +----+------+----------+------+----------+
           | 1  |  1   | 1 to 255 |  1   | 1 to 255 |
           +----+------+----------+------+----------+
```

### CODE REVIEW

File: transport/socks5/socks5.go [hunk 1]
```diff
@@ -195,17 +195,29 @@ func ClientHandshake(rw io.ReadWriter, addr Addr, command Command, user *User) (
...
 
-		authMsgLen := 1 + 1 + len(user.Username) + 1 + len(user.Password)
-		if authMsgLen > MaxAuthLen {
-			return nil, errors.New("auth message too long")
+		uLen := len(user.Username)
+		pLen := len(user.Password)
+
+		// Both ULEN and PLEN are limited to the range [1, 255].
+		switch {
+		case uLen == 0:
```
Comment (xjasonlyu) [inline transport/socks5/socks5.go:203]: I think here we can simply check if username/password is missing or too long, no need to check them separately.

**Review by xjasonlyu — CHANGES_REQUESTED** [review]
Thanks for your PR!

Just some minor stuff:

File: transport/socks5/socks5.go (hunk 1 above)
Comment (Yawning) [reply transport/socks5/socks5.go]: Fixed, I hope.

Comment (xjasonlyu) [reply transport/socks5/socks5.go]: Can be this:

```go
if uLen == 0 || pLen == 0 {
	return nil, errors.New("auth username/password empty")
} else if uLen > MaxAuthLen || pLen > MaxAuthLen {
	return nil, errors.New("auth username/password too long")
}
```

Comment (Yawning) [reply transport/socks5/socks5.go]: Done.

**Comment by xjasonlyu** [comment]
Merged, thanks!

**Pull request merged**